### PR TITLE
Pase environment variables

### PR DESCRIPTION
**Pull request by worksofliam**
### Changes

Currently, variables are only replaced before the command is executed. This PR adds local variables as environment variables for pase Actions.

Implements #896

#### Sample

```jsonc
// action
        {
            "type": "member",
            "extensions": [
                "GLOBAL"
            ],
            "environment": "pase",
            "name": "Testing envvar",
            "command": "echo \"$CURLIB $USERNAME $LIBLC\""
        }
```

```
# result
Current library: LIAMA
Library list: SAMPLE QDEVTOOLS
Commands:
		echo "$CURLIB $USERNAME $LIBLC"

LIAMA linux QDEVTOOLS,SAMPLE
```

### To do

* [ ] Acceptance testing
* [x] Docs updated: https://halcyon-tech.github.io/docs/#/pages/developing/local/getting-started?id=pase-environment-variables

### Checklist

* [x] have tested my change
* [x] updated relevant documentation
* [ ] Remove any/all `console.log`s I added
* [ ] eslint is not complaining
* [ ] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/halcyon-tech/vscode-ibmi/blob/master/CONTRIBUTING.md)
* [x] **for feature PRs**: PR only includes one feature enhancement.

### CODE REVIEW

**Comment by worksofliam** [comment]
@Wright4i @chrjorgensen Any chance either of you could test this out for me?

**Comment by Wright4i** [comment]
@worksofliam sure thing. I'll test it tomorrow morning!

**Comment by Wright4i** [comment]
Working as expected on my end!

**Comment by chrjorgensen** [comment]
@worksofliam I am about to test this - and had a thought about referencing the variables in the command: It may not be obvious that any variable should be referenced as $VAR when the action screen shows &VAR... Maybe an implicit conversion could be done so the user can specify &VAR as usual and this will be converted to $VAR when sent to PASE?

**Comment by worksofliam** [comment]
@chrjorgensen i believe that issue is coming from the RPGLE language tools extension and will be fixed in the next release for that (as I just fixed it).

I assume you are still able to save and compile actions like normal?

**Comment by worksofliam** [comment]
@chrjorgensen 

> It may not be obvious that any variable should be referenced as $VAR when the action screen shows &VAR

The user will still be able to use `&MYVAR` and that's how we will keep recommending it on the UI (due to a magnitude of reasons). The ability to use the environment variables is something we will document but not mention on the UI yet.

This PR is really to enable the programs that are being called to be able to access these variables as well, not so much from the Action command.

**Comment by chrjorgensen** [comment]
@worksofliam 

> I assume you are still able to save and compile actions like normal?

Yes, no problem in saving actions - only this "error" in the output screen, when saving an ILE action.

**Comment by worksofliam** [comment]
@chrjorgensen I am going to look into that as a separate issue today. Thanks!

**Comment by worksofliam** [comment]
@halcyon-tech/core  Some further cleanup on this PR today. Any chance we could do some further testing?

Updated docs page here: https://halcyon-tech.github.io/docs/#/pages/developing/local/getting-started?id=pase-environment-variables

**Comment by worksofliam** [comment]
I've done further testing and all existing functionality works as expected and new functionality works as expected also.

File: src/api/CompileTools.js [hunk 1]
```diff
@@ -466,12 +467,20 @@ module.exports = class CompileTools {
 
           outputBarItem.text = OUTPUT_BUTTON_RUNNING;
 
+          if (workspaceFolder) {
+            const envFileVars = await getEnvConfig(workspaceFolder);
+            Object.entries(envFileVars).forEach(item => {
+              variables[`&` + item[0]] = item[1];
```
Comment (worksofliam) [inline src/api/CompileTools.js:473]: This makes `.env` variables accessible on the Action command also.

File: src/api/CompileTools.js [hunk 2]
```diff
@@ -633,9 +641,20 @@ module.exports = class CompileTools {
 
       switch (options.environment) {
       case `pase`:
+        // We build environment variables for the environment to be ready
+        /** @type {{[name: string]: string}} */
+        const envVars = {};
+        Object
+          .entries({...(options.env ? options.env : {}), ...this.getDefaultVariables(instance)})
+          .filter(item => (new RegExp(`^[A-Za-z\&]`, `i`).test(item[0])))
```
Comment (worksofliam) [inline src/api/CompileTools.js:649]: Only accept environment variables that start with a letter or `&`.

File: src/api/CompileTools.js [hunk 3]
```diff
@@ -633,9 +641,20 @@ module.exports = class CompileTools {
...
+          .forEach(item => {
+            envVars[item[0][0] === `&` ? item[0].substring(1) : item[0]] = item[1];
```
Comment (worksofliam) [inline src/api/CompileTools.js:651]: If it starts with `&`, remove it.

File: src/api/IBMi.ts [hunk 4]
```diff
@@ -657,7 +649,16 @@ export default class IBMi {
   }
 
   async sendCommand(options: CommandData): Promise<CommandResult> {
-    const command = options.command;
+    let commands = [];
+    if (options.env) {
+      commands.push(...Object.keys(options.env).map(envVar => `export ${envVar}="${
+        options.env ? options.env[envVar].replace(/\$/g, `\\$`).replace(/"/g, `\\"`) : ``
```
Comment (worksofliam) [inline src/api/IBMi.ts:655]: Build up the `export` commands to prepend the executing command.

**Pull request merged**